src/redfish_collector/core/rawCollector.py

```
@@ -9,14 +9,24 @@
 
 
 
 def readYAMLTemplate(templateFile, serverAddress):
+    # Resolve a single path and use it for BOTH the existence check and the open.
+    # path.join ignores the module dir when templateFile is already absolute
+    # (the service passes absolute paths), so production behaviour is unchanged;
+    # this only fixes the relative-path case where isfile and open disagreed.
     config_file_path = path.join(path.dirname(__file__), templateFile)
     if path.isfile(config_file_path):
-        with open(templateFile, 'r') as f:
+        with open(config_file_path, 'r') as f:
```

Ruff (0.15.18)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)